.github/workflows/publish-release.yml

```
@@ -47,23 +66,11 @@ jobs:
                     publishUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
                     publishPAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
                 run: |
-                    ./gradlew release -Prelease.useAutomaticVersion=true
-                    ./gradlew -Pversion=${VERSION} publish -x test
-
-            -   name: Create Github Release
-                run: |
-                    curl --request POST 'https://api.github.com/repos/ballerina-platform/module-ballerina-serdes/releases' \
-                    --header 'Accept: application/vnd.github.v3+json' \
-                    --header 'Authorization: Bearer ${{ secrets.BALLERINA_BOT_TOKEN }}' \
-                    --header 'Content-Type: application/json' \
-                    --data-raw '{
-                        "tag_name": "v'"$VERSION"'",
-                        "name": "module-ballerina-serdes-v'"$VERSION"'"
-                    }'
-
-            -   name: Create Post release PR
+                    ./gradlew clean release -Prelease.useAutomaticVersion=true
+                    ./gradlew -Pversion=${VERSION} publish -x test -PpublishToCentral=true
+            -   name: GitHub Release and Release Sync PR
                 env:
                     GITHUB_TOKEN: ${{ secrets.BALLERINA_BOT_TOKEN }}
                 run: |
-                    curl -fsSL https://github.com/github/hub/raw/master/script/get | bash -s 2.14.1
-                    bin/hub pull-request --base main -m "[Automated] Sync main after "$VERSION" release"
+                    gh release create v$VERSION --title "module-ballerina-serdes-v$VERSION"
+                    gh pr create --title "[Automated] Sync master after $VERSION release" --body "Sync master after $VERSION release"
```

Since we have a `main` branch, we have to explicitly mention the base of the PR.

```suggestion
                    gh pr create --base main --title "[Automated] Sync main after $VERSION release" --body "Sync main after $VERSION release"
```